src/osd/PrimaryLogPG.cc

```diff
@@ -1756,6 +1756,11 @@ PrimaryLogPG::PrimaryLogPG(OSDService *o, OSDMapRef curmap,
   m_scrubber = make_unique<PrimaryLogScrub>(this);
 }
 
+PrimaryLogPG::~PrimaryLogPG()
+{
+  m_scrubber.reset();
```

Comment:
per https://tracker.ceph.com/issues/51843, `PgScrubber::rm_from_osd_scrubbing()` is called by the dtor of `PgScrubber `. but i cannot find where `PgScrubber::rm_from_osd_scrubbing()` is defined or gets called. could you be more specific on

> we guarantee that Scrubber's code can refer to the PG object - especially in dout()s.

what the Scrubber's code is exactly?

also, IMHO, no need to reference #42479 in the commit message. it should be put in the comment, not the commit message.

Reply:
Agree and removed the ref in the commit message.

Re your other questions:
I have a bunch of PRs that either fix scrub issues or advance scrub code refactoring (and towards Crimson). Most of them are in flight, waiting either approval or final testing (and one "draft" bug fix). Testing each of those individually is tiresome, as I have to analyze test failures caused by the other known issues (fixed by the other PRs).

I am now PRing some of the simple bug fixes separately, both to speed the checking of the rest of the PRs, and for some - to facilitate a simple backporting.

For the specific comments in 51843: the code there is from PR #40984. But such failures were encountered in merged code, too. After some discussions with @athanatos, one of the required changes was to early-destruct the Scrubber ('scrubber' here - PgScrubber and it sub-objects).

Reply:
the commit message reads

> Note: extracted from PR#42479, to facilitate backporting.

i assume that you want to backport this PR. if that's the case, my question still holds.

Reply:
> Agree and removed the ref in the commit message.

the commit message has not been changed.

Reply:
Oops. Not it is.

Reply:
> i assume that you want to backport this PR. if that's the case, my question still holds.

not sure I understand

Reply:
i mean, my question is not addressed. because if the issue only exists along with "a bunch of PRs", then we don't need to backport this change.

Reply:
Got it.

The problem did happen in existing code, without the PRs. I didn't save specific logs, but (not specifically for this PR) I am trying to recreate those crashes.

And - this should be considered when deciding whether to backport. For 'master' - the change is important by itself.